### PR TITLE
fix to get default storage from cookie

### PR DESCRIPTION
**Pull request by daruj**

### CODE REVIEW

File: src/generators/app/templates/src/client/index.jsx [hunk 1]
```diff
@@ -15,6 +16,9 @@ if (config.get('api.fixtures')) {
 
 const store = createStore(storageEngine);
 
+const load = storage.createLoader(storageEngine);
+load(store);
```
Comment (erichulburd) [inline src/generators/app/templates/src/client/index.jsx]: creo que nada mas `storage.createLoader(storageEngine)(store);` - es mas leible.

File: src/generators/app/templates/src/server/renderReact.jsx [hunk 2]
```diff
@@ -15,6 +16,8 @@ import createStore from 'shared/redux/createStore';
 export default function renderReact(req) {
   const storageEngine = createEngine(config.get('storage.key'), req);
   const store = createStore(storageEngine);
+  const load = storage.createLoader(storageEngine);
```
Comment (erichulburd) [inline src/generators/app/templates/src/server/renderReact.jsx]: tambien aqui: `storage.createLoader(storageEngine)(store);`.

File: src/generators/app/templates/src/server/configureApp.js [hunk 3]
```diff
@@ -29,18 +32,31 @@ function handleErr(res, err) {
...
     if (config.get('isomorphic')) {
-      const {
-        renderedContent
-      } = renderReact(req);
-      content = renderedContent;
+      const storageEngine = createEngine(config.get('storage.key'), req);
+      const store = createStore(storageEngine);
+      storage.createLoader(storageEngine)(store)
+        .then(() => {
+          content = renderReact(req.i18n, req.originalUrl, {}, store);
```
Comment (erichulburd) [inline src/generators/app/templates/src/server/configureApp.js]: `content = renderReact(req.i18n, req.originalUrl, context, store);`

File: src/generators/app/templates/src/server/renderReact.jsx [hunk 4]
```diff
@@ -1,35 +1,22 @@
...
-  const storageEngine = createEngine(config.get('storage.key'), req);
-  const store = createStore(storageEngine);
-  const redirect = {};
-  const renderedContent = ReactDOMServer.renderToString(
-    <Provider store={store}>
-      <StaticRouter location={req.url} context={redirect}>
-        <Layout i18n={req.i18n} />
-      </StaticRouter>
-    </Provider>
+export default function renderReact(i18n, location, context = {}, store) {
```
Comment (erichulburd) [inline src/generators/app/templates/src/server/renderReact.jsx]: `export default function renderReact(i18n, location, context, store) {`

File: src/generators/app/templates/src/shared/routes/CustomRoute.jsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Route, withRouter, Redirect } from 'react-router';
+import { connect } from 'react-redux';
+
+export const RouteWithStatus = ({ code, children }) => (
+  <Route
+    render={({ staticContext }) => {
+      if (staticContext) {
+        staticContext.status = code;
```
Comment (erichulburd) [inline src/generators/app/templates/src/shared/routes/CustomRoute.jsx]: Para que sirve esto? Pense que ReactRouter hace esto automaticamente?

**Review by erichulburd — CHANGES_REQUESTED** [review]

File: src/generators/app/templates/src/shared/routes/CustomRoute.jsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+);
+
+RedirectWithStatus.propTypes = {
+  from: PropTypes.string.isRequired,
+  to: PropTypes.string.isRequired,
+  status: PropTypes.number.isRequired
+};
+
+
+const CustomRoute = ({
```
Comment (erichulburd) [inline src/generators/app/templates/src/shared/routes/CustomRoute.jsx]: Creo que necesita un nombre mas especifico - `SwitchRoute`, `AuthConfiguredRoute`, etc.

File: src/generators/app/templates/src/shared/routes/CustomRoute.jsx (hunk 5 above)
Comment (daruj) [reply src/generators/app/templates/src/shared/routes/CustomRoute.jsx]: there is no `staticContext` on the client, so we need to guard against that here

**Pull request merged**